website_sharing_whole_class_mosaic_plot_translation.Rmd

```diff
@@ -0,0 +1,507 @@
+* ddecker()
+ + 选择n-1个水平spine，并以竖向spine结束
+* 定义每种类型的分区
+ + c("hspine"、"vspine"、"hbar")
+ 
+```{r}
+h_mosaic <- ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq, Gender, Region), fill=FlightFreq), na.rm=T, divider=mosaic("h")) +
+  theme(axis.text.x=element_blank(), legend.position="none") + 
+  labs(x=" ", title='divider= mosaic()')
+
+v_mosaic <- ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq, Gender, Region), fill=FlightFreq), na.rm=T, divider=mosaic("v")) +
+  theme(axis.text.x=element_blank()) + 
+  labs(x=" ", title='divider= mosaic("v")')
+
+
+doubledecker <- ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq, Gender, Region), fill=FlightFreq), na.rm=T, divider=ddecker()) +
+  theme(axis.text.x=element_blank()) +
+  labs(x=" ", title='divider= ddecker()')
+
+h_mosaic
+v_mosaic
+doubledecker
+
+```
+ 
+
+```{r}
+mosaic4 <-  ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq, Gender, Region), fill=FlightFreq), na.rm=T, divider=c("vspine", "vspine", "hbar")) +
+  theme(axis.text.x=element_blank()) +
+  labs(x=" ", title='divider= c("vspine", "vspine", "hbar")') 
+
+mosaic5 <-  ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq, Gender, Region), fill=FlightFreq), na.rm=T, divider=c("hbar", "vspine", "hbar")) +
+  theme(axis.text.x=element_blank()) +
+  labs(x=" ", title='divider= c("hbar", "vspine", "hbar")')
+
+mosaic6 <-  ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq, Gender, Region), fill=FlightFreq), na.rm=T, divider=c("hspine", "hspine", "hspine")) +
+  theme(axis.text.x=element_blank()) +
+  labs(x=" ", title='divider= c("hspine", "hspine", "hspine")')
+
+mosaic7 <-  ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq, Gender, Region), fill=FlightFreq), na.rm=T, divider=c("vspine", "vspine", "vspine")) +
+  theme(axis.text.x=element_blank()) +
+  labs(x=" ", title='divider= c("vspine", "vspine", "vspine")')
+
+mosaic4
+mosaic5
+mosaic6
+mosaic7
+```
+
+
+### 3.9 geom_mosaic: offset
+
+offset。设置第一条spine之间的空间大小
+
+* 默认值default=0.01
+* 分区之间的空间会随着层数的增加而减小。
+
+### 3.10 调整offset
+
+```{r}
+offset1 <- ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq, Region), fill=FlightFreq), na.rm=TRUE) + labs(x="Region", y=" ",  title=" offset = 0.01") 
+
+offset0 <- ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq, Region), fill=FlightFreq), na.rm=TRUE, offset = 0) + labs(x="Region", y=" ",  title=" offset = 0")
+
+offset2 <- ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq, Region), fill=FlightFreq), na.rm=TRUE, offset = 0.02) + labs(x="Region", y=" ",  title=" offset = 0.02")
+
+offset1
+offset0
+offset2
+```
+
+
+### 3.11 Plotly
+
+```{r}
+library(plotly)
+gg <-ggplot(data = fly) +
+   geom_mosaic(aes(x = product(DoYouRecline, RudeToRecline), fill=DoYouRecline), na.rm=TRUE) + labs(x = "Is it rude recline? ", title='f(DoYouRecline | RudeToRecline) f(RudeToRecline)')
+ggplotly(gg)
+```
+
+
+
+## 4 参考文献
```

Comment:
Hashtag headers should not be followed by numbers since the hashtags will create numbered subheadings.